src/python/pants/core/goals/test.py

```diff
@@ -204,6 +226,149 @@ class TestFieldSet(FieldSet, metaclass=ABCMeta):
     __test__ = False
 
 
+_T = TypeVar("_T")
+_FieldSetT = TypeVar("_FieldSetT", bound=TestFieldSet)
+
+
+@runtime_ignore_subscripts
+class Partitions(FrozenDict[Any, "tuple[_FieldSetT, ...]"]):
```

Comment:
I'm using `Partition` for internal / plugin API names  and `batch_*` for the user-facing API - we have the same dichotomy for `lint`/`fmt` so it's hopefully not confusing, but I'm not attached to any names used in the new code